src/app/pages/checkout-payment/checkout-payment/checkout-payment.component.ts

```diff
@@ -46,8 +36,9 @@ export class CheckoutPaymentComponent implements OnInit, OnChanges, OnDestroy {
   @Input() error: HttpError;
 
   @Output() updatePaymentMethod = new EventEmitter<string>();
-  @Output() createPaymentInstrument = new EventEmitter<PaymentInstrument>();
-  @Output() deletePaymentInstrument = new EventEmitter<string>();
+  @Output() createBasketPaymentInstrument = new EventEmitter<PaymentInstrument>();
+  @Output() createUserPaymentInstrument = new EventEmitter<PaymentInstrument>();
```

Comment:
Splitting the output into `createBasketPaymentInstrument` and `createUserPaymentInstrument` seems unnecessary, especially since it all goes into the same facade action in the end anyways. So only one `createPaymentInstrument` with the additional 'save' information seems cleaner.
Needs to be changed here and the according other places.